File: KITE-Simulcast-Test/src/main/java/io/cosmosoftware/kite/simulcast/steps/GetStatsStep.java
package io.cosmosoftware.kite.simulcast.steps;

import io.cosmosoftware.kite.exception.KiteTestException;
import io.cosmosoftware.kite.interfaces.Runner;
import io.cosmosoftware.kite.report.Reporter;
import io.cosmosoftware.kite.report.Status;
import io.cosmosoftware.kite.steps.TestStep;
import java.util.List;
import org.openqa.selenium.WebDriver;

import javax.json.JsonObject;
import org.webrtc.kite.stats.RTCStatList;
import org.webrtc.kite.stats.RTCStats;

import static org.webrtc.kite.stats.StatsUtils.getPCStatOvertime;
import static org.webrtc.kite.stats.StatsUtils.transformToJson;

public class GetStatsStep extends TestStep {

  private final JsonObject getStatsConfig;

  public GetStatsStep(Runner runner, JsonObject getStatsConfig) {
    super(runner);
    this.getStatsConfig = getStatsConfig;
  }

  
  
  @Override
  public String stepDescription() {
    return "GetStats";
  }
  
  @Override
  protected void step() throws KiteTestException {
    try {
      RTCStatList localPcStats = getPCStatOvertime(webDriver, getStatsConfig).get(0);
      reporter.jsonAttachment(report, "getStatsRaw", transformToJson(localPcStats));
    } catch (Exception e) {
      e.printStackTrace();
      throw new KiteTestException("Failed to getStats", Status.BROKEN, e);
    }
  }
}

File: KITE-Jitsi-Test/js/steps/GetStatsStep.js
const {TestUtils, TestStep, Status, KiteTestError} = require('kite-common');

class GetStatsStep extends TestStep {
  constructor(kiteBaseTest) {
    super();
    this.driver = kiteBaseTest.driver;
    this.statsCollectionTime = kiteBaseTest.statsCollectionTime;
    this.statsCollectionInterval = kiteBaseTest.statsCollectionInterval;
    this.selectedStats = kiteBaseTest.selectedStats;
    this.page = kiteBaseTest.page;
    this.peerConnections = kiteBaseTest.peerConnections;

    // Test reporter if you want to add attachment(s)
    this.testReporter = kiteBaseTest.reporter;
  }

  stepDescription() {
    return 'Getting WebRTC stats via getStats';
  }

  async step() {
    try {
    let rawStats = await this.page.getStats(this);
    let summaryStats = TestUtils.extractJson(rawStats, 'both');
    // // Data
    this.testReporter.textAttachment(this.report, 'GetStatsRaw', JSON.stringify(rawStats, null, 4), "json");
    this.testReporter.textAttachment(this.report, 'GetStatsSummary', JSON.stringify(summaryStats, null, 4), "json");
  } catch (error) {
      console.log(error);
      throw new KiteTestError(Status.BROKEN, "Failed to getStats");
    }
  }
}

module.exports = GetStatsStep;